src/views/Transitioner.js

```diff
@@ -58,12 +53,15 @@ class Transitioner extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const nextScenes = NavigationScenesReducer(
+    let nextScenes = NavigationScenesReducer(
       this.state.scenes,
       nextProps.navigation.state,
       this.props.navigation.state,
       nextProps.descriptors
     );
+    if (!nextProps.navigation.state.isTransitioning) {
+      nextScenes = filterStale(nextScenes);
```

Comment:
here's the only behavioral change. Now we will immediately get rid of stale scenes unless a transition has been explicitly requested.